src/lib/layer.js

```diff
@@ -446,7 +446,10 @@ export default class Layer {
     settings.polygonOffset = offsets;
 
     // Call subclass lifecycle method
-    this.draw({moduleParameters, uniforms, settings});
+    withParameters(this.context.gl,
```

Comment:
Also `settings` is not documented - do we have a good use case for it? If yes, maybe should call it `glParameters` to be clear?